client/my-sites/importer/site-importer/url-validation.js

```diff
@@ -0,0 +1,47 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import url from 'url';
+import { isWebUri } from 'valid-url';
+import { translate } from 'i18n-calypso';
+import { inRange, trim } from 'lodash';
+
+const parseUrl = function( value = '' ) {
+	const parsedUrl = url.parse( value, false, true );
```

Comment:
Do we need to wrap the `url.parse` in a try / catch? It throws `TypeError` & `URIError`.